docs: Adding Interactivity Overview translations (#350)
reactjs/id.react.dev: src/content/learn/adding-interactivity.md

---
title: Menambahkan Interaktivitas
---

<Intro>

> Beberapa hal di layar berubah mengikuti input dari pengguna. Contohnya, mengeklik sebuah galeri gambar bisa mengganti gambar yang sedang aktif. Di React, data yang berubah seiring waktu disebut *state.* Anda dapat menambahkan state ke komponen apapun, dan mengubahnya sesuai kebutuhan. Di bab ini, Anda akan belajar cara menulis komponen yang dapat menangani interaksi, mengubah state yang dimilikinya, dan menampilkan output yang berbeda seiring berjalannya waktu.

Reviewer: Input -> Masukan
Output -> Keluaran 
```suggestion
Beberapa hal di layar berubah mengikuti masukan dari pengguna. Contohnya, mengeklik sebuah galeri gambar bisa mengganti gambar yang sedang aktif. Di React, data yang berubah seiring waktu disebut *state.* Anda dapat menambahkan state ke komponen apapun, dan mengubahnya sesuai kebutuhan. Di bab ini, Anda akan belajar cara menulis komponen yang dapat menangani interaksi, mengubah state yang dimilikinya, dan menampilkan keluaran yang berbeda seiring berjalannya waktu.
```

</Intro>

<YouWillLearn isChapter={true}>

* [Cara menangani events yang di-mulai oleh pengguna](/learn/responding-to-events)
* [Cara membuat komponen mengingat informasi dengan menggunakan state](/learn/state-a-components-memory)
* [Cara React memperbaharui UI dalam dua fase](/learn/render-and-commit)
* [Mengapa state tidak langsung terbaharui setelah Anda mengubahnya](/learn/state-as-a-snapshot)

> * [Cara meng-queue beberapa perubahan state](/learn/queueing-a-series-of-state-updates)

Reviewer: ```suggestion
* [Cara mengantrikan beberapa perubahan state](/learn/queueing-a-series-of-state-updates)
```

> * [Cara mengubah object didalam state](/learn/updating-objects-in-state)

Reviewer: ```suggestion
* [Cara mengubah object di dalam state](/learn/updating-objects-in-state)
```

> * [Cara mengubah array didalam state](/learn/updating-arrays-in-state)

Reviewer: ```suggestion
* [Cara mengubah array di dalam state](/learn/updating-arrays-in-state)
```

</YouWillLearn>

## Menanggapi events {/*responding-to-events*/}

React memungkinkan Anda untuk menambakan *event handlers* ke JSX Anda. *Event handlers* adalah fungsi milik Anda yang akan dipanggil sebagai respon terhadap interaksi dari pengguna seperti klik, *hover*, fokus pada input form, dan lain-lain.

> Komponen bawaan seperti `<button>` hanya mendukung *event* bawaan dari browser seperti `onClick`. Namun, Anda juga dapat membuat komponen Anda sendiri, dan memberikannya *prop event handler* dengan nama apapun, spesifik terhadap aplikasi Anda.

Reviewer: ```suggestion
Komponen bawaan seperti `<button>` hanya mendukung *event* bawaan dari peramban seperti `onClick`. Namun, Anda juga dapat membuat komponen Anda sendiri, dan memberikannya *prop event handler* dengan nama apapun, spesifik terhadap aplikasi Anda.
```

<LearnMore path="/learn/state-a-components-memory">

Baca **[State: Ingatan dari komponen](/learn/state-a-components-memory)** untuk mempelajari cara untuk mengingat sebuah nilai dan mengubahnya saat berinteraksi.

</LearnMore>

## Render dan commit {/*render-and-commit*/}

Sebelum komponen Anda ditampilkan di layar, mereka harus di *render* oleh React. Memahami langkah-langkah dalam proses ini akan membantu Anda berpikir tentang bagaimana kode Anda akan dijalankan dan menjelaskan perilakunya.

Bayangkan bahwa komponen Anda adalah koki di dapur, menyusun hidangan lezat dari bahan-bahan dasar. Dalam skenario ini, React adalah pelayan yang mengajukan permintaan dari pelanggan dan membawakan pesanan mereka. Proses permintaan dan pelayanan UI ini memiliki tiga langkah:

1. **Memicu** *render* (mengirimkan pesanan pelanggan ke dapur)

> 2. **Merender** komponen (menyiapkan pesanan di dapur)

Reviewer 2: Once more. We have been updated the terms and for word like "merender" need to changes "me-*render*" except in **Heading** ("me-render").

Author: Thanks, updated.

3. **Commit** ke DOM (menempatkan pesanan di meja)

<IllustrationBlock sequential>
  <Illustration caption="Memicu" alt="React sebagai pelayan di restoran, mengambil pesanan dari pengguna dan mengantarkannya ke Dapur Komponent" src="/images/docs/illustrations/i_render-and-commit1.png" />
  <Illustration caption="Merender" alt="Koki Card memberikan React komponen Card baru." src="/images/docs/illustrations/i_render-and-commit2.png" />
  <Illustration caption="Commit" alt="React mengantarkan Card ke pengguna di meja mereka" src="/images/docs/illustrations/i_render-and-commit3.png" />
</IllustrationBlock>

<LearnMore path="/learn/render-and-commit">

Baca **[Render dan Commit](/learn/render-and-commit)** untuk mempelajari *lifecycle* dari perubahan UI.

</LearnMore>

## State sebagai snapshot {/*state-as-a-snapshot*/}

> Tidak seperti variabel Javascript biasa, state di React berperilaku lebih seperti *snapshot*. Mengubah state tidaklah mengubah variabel state yang Anda miliki sekarang, tetapi akan memicu re-render. Ini bisa mengejutkan pada awalnya!

Reviewer: Semua re-render atau rerender = *render* ulang.
Dengan italic di render.

```suggestion
Tidak seperti variabel Javascript biasa, state di React berperilaku lebih seperti *snapshot*. Mengubah state tidaklah mengubah variabel state yang Anda miliki sekarang, tetapi akan memicu *render* ulang. Ini bisa mengejutkan pada awalnya!
```

Author: Thanks, updated.

<LearnMore path="/learn/state-as-a-snapshot">

Baca **[State sebagai Snapshot](/learn/state-as-a-snapshot)** untuk mempelajari mengapa state terlihat "tetap" dan tak berubah di dalam *event handler*.

</LearnMore>

> ## Meng-queue serangkaian perubahan state {/*queueing-a-series-of-state-updates*/}

Reviewer: ```suggestion
## Mengantrikan serangkaian perubahan state {/*queueing-a-series-of-state-updates*/}
```

console.log(score);  // 0
```

> Anda bisa memperbaiki ini dengan memberikan *updater function* ketika mengubah state. Perhatikan bagaimana mengganti `setScore(score + 1)` dengan `setScore(s => s + 1)` memperbaiki tombol "+3". Cara ini memungkinkan Anda untuk meng-*queue* beberapa perubahan state.

Reviewer: ```suggestion
Anda bisa memperbaiki ini dengan memberikan *updater function* ketika mengubah state. Perhatikan bagaimana mengganti `setScore(score + 1)` dengan `setScore(s => s + 1)` memperbaiki tombol "+3". Cara ini memungkinkan Anda untuk mengantrikan beberapa perubahan state.
```

<LearnMore path="/learn/queueing-a-series-of-state-updates">

> Baca **[Meng-queue serangkaian perubahan state](/learn/queueing-a-series-of-state-updates)** untuk mempelajari cara meng-*queue* sebuah rentetan perubahan state.

Reviewer: ```suggestion
Baca **[Mengantrikan serangkaian perubahan state](/learn/queueing-a-series-of-state-updates)** untuk mempelajari cara meng-*queue* sebuah rentetan perubahan state.
```

<LearnMore path="/learn/updating-objects-in-state">

> Baca **[Mengubah Object didalam State](/learn/updating-objects-in-state)** untuk mempelajari cara mengubah *object* dengan benar.

Reviewer: ```suggestion
Baca **[Mengubah Object di dalam State](/learn/updating-objects-in-state)** untuk mempelajari cara mengubah *object* dengan benar.
```

</LearnMore>

> ## Mengubah array didalam state {/*updating-arrays-in-state*/}

Reviewer: ```suggestion
## Mengubah array di dalam state {/*updating-arrays-in-state*/}
```

<LearnMore path="/learn/updating-arrays-in-state">

> Baca **[Mengubah Array didalam State](/learn/updating-arrays-in-state)** untuk mempelajari cara mengubah *array* dengan benar.

Reviewer: ```suggestion
Baca **[Mengubah Array di dalam State](/learn/updating-arrays-in-state)** untuk mempelajari cara mengubah *array* dengan benar.
```
